code/modules/mob/living/blood.dm

```diff
@@ -160,6 +166,9 @@
 	if(!O.reagents)
 		return
 
+	if(blood_volume < amount)
+		amount = blood_volume
```

Comment:
should probably sanity check for blood volume here so you don't go through the entire proc with `amount` being zero, same for /carbon